src/pages/ChatPage.ts

```diff
@@ -0,0 +1,258 @@
+		const loginAddInput = new InputBlock(
+			undefined,
+			{
+				input: {attr: {type: 'text', name: fieldLoginName, value: '', placeholder: ' ', autofocus: true}},
+				name: fieldLoginName,
+				label: 'Логин',
+				valid: REG_EXP.LOGIN, 
+				fieldErrorMsg: ERROR_MSG.LOGIN
+			}
+		);
+		const loginDelInput = new InputBlock(
+			undefined,
+			{
+				input: {attr: {type: 'text', name: fieldLoginName, value: '', placeholder: ' ', autofocus: true}},
+				name: fieldLoginName,
+				label: 'Логин',
+				valid: REG_EXP.LOGIN, 
+				fieldErrorMsg: ERROR_MSG.LOGIN
+			}
+		);
+		const addUserBtn = new Button(undefined,
+			{
+				label: 'Добавить',
+				attr: {type: 'submit', id: 'addUserToChat'}
+			});
+		const addUserForm = new Form(undefined,
+			{
+				// formClass: 'form',
+				title: 'Добавить пользователя',
+				inputs: new List(undefined, {loginAddInput}),
+				button: addUserBtn,
+				request: {
+					f_submit: userApi.searchUsersByLogin,
+					resolve: (resp: string)=>{
+						toggleAddUserModal(false);
+						resp = JSON.parse(resp);
+						if(Array.isArray(resp)){
+							const users = resp.map(user=>user.id),
+								chatId = this.children?.chatPage?.children?.main?.props?.chatInfo?.id;
+							
+							chatApi.addUser(users, chatId).catch(err=>{
+								console.log('addUser: err='+typeof err, err);
+							});
+						}
+					},
+					reject: (err: Error)=>{
+						console.log('err='+typeof err, err);
+						this.setProps({...this.props, errorMsg: 'Server error'});
+					}
+				}
+			},
+			'modal__dialog');
+		const modalAddUser = new Modal(
+			undefined,
+			{
+				form: addUserForm
+			},
+			'modal'
+		);
+		const deleteUserBtn = new Button(undefined,
+			{
+				label: 'Удалить',
+				attr: {type: 'submit', id: 'deleteUserToChat'}
+			});
+		const deleteUserForm = new Form(undefined,
+			{
+				// formClass: 'form',
+				title: 'Удалить пользователя',
+				inputs: new List(undefined, {loginDelInput}),
+				button: deleteUserBtn,
+				request: {
+					f_submit: userApi.searchUsersByLogin,
+					resolve: (resp: string)=>{
+						toggleDeleteUserModal(false);
+						resp = JSON.parse(resp);
+						if(Array.isArray(resp)){
+							const users = resp.map(user=>user.id),
+								chatId = this.children?.chatPage?.children?.main?.props?.chatInfo?.id;
+							
+							chatApi.deleteUser(users, chatId).catch(err=>{
+								console.log('deleteUser: err='+typeof err, err);
+							});
+						}
+					},
+					reject: (err: Error)=>{
+						console.log('err='+typeof err, err);
+						this.setProps({...this.props, errorMsg: 'Server error'});
+					}
+				}
+			},
+			'modal__dialog');
+		const modalDeleteUser = new Modal(
+			undefined,
+			{
+				form: deleteUserForm
+			},
+			'modal'
+		);
+
+		const toggleAddUserModal = (mode: boolean) =>{
+			if(mode){
+				modalAddUser.show();
+			} else if(mode === false){
+				modalAddUser.hide();
+			} else {
+				modalAddUser.isShow ? modalAddUser.hide() : modalAddUser.show();
+			}
```

Comment:
Вот тут не очень понял. Если mode: boolean, то есть всего 2 пути.
```suggestion
			if(!mode){
			    modalAddUser.hide();
			    return;
			}
			   modalAddUser.show()
		           modalAddUser.isShow ? modalAddUser.hide() : modalAddUser.show();
			}
```

Reply:
Там был вариант с underfined. Но я это потом переделал. Я исправил. Убрал лишнее